templates/routing-informs/encoder.go

```diff
@@ -0,0 +1,244 @@
+package main
+
+import (
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/nextmv-io/sdk"
+	"github.com/nextmv-io/sdk/route"
+	"github.com/nextmv-io/sdk/run"
+	"github.com/nextmv-io/sdk/run/encode"
+)
+
+type output struct {
+	Store      route.Plan   `json:"store"`
+	Statistics statisticsIn `json:"statistics"`
+}
+
+// statisticsIn of the search.
+type statisticsIn struct {
+	Time Time `json:"time"`
+	// Value of the store. Nil when using a Satisfier.
+	Value *int `json:"value,omitempty"`
+}
+
+// Time needed.
+type Time struct {
+	Start   time.Time `json:"start"`
+	Elapsed duration  `json:"elapsed"`
+}
+
+type duration struct {
+	time.Duration
+}
+
+func (d duration) MarshalJSON() ([]byte, error) {
+	return json.Marshal(d.String())
+}
+
+func (d *duration) UnmarshalJSON(b []byte) error {
+	var v interface{}
+	if err := json.Unmarshal(b, &v); err != nil {
+		return err
+	}
+	switch value := v.(type) {
+	case float64:
+		d.Duration = time.Duration(value)
+		return nil
+	case string:
+		var err error
+		d.Duration, err = time.ParseDuration(value)
+		if err != nil {
+			return err
+		}
+		return nil
+	default:
+		return errors.New("invalid duration")
+	}
+}
+
+type statisticsOut struct {
+	Schema string `json:"schema"`
+	Run    struct {
```

Comment:
Is `statisticsOut.Run.Time` set somewhere? Otherwise this is always 0. You could also leave out `Run`, as it is optional.

Reply:
this is the runtime that was initally given, right? then, I can kill it.
And the schema should be `v1`?